src/WebJobs.Extensions.OpenAI/Search/SemanticSearchAttribute.cs

```diff
@@ -40,14 +40,24 @@ public SemanticSearchAttribute(string connectionName, string collection)
     public string ConnectionName { get; set; }
 
     /// <summary>
-    /// The name of the collection or table to search.
+    /// The name of the collection or table or index to search.
     /// </summary>
     /// <remarks>
     /// This property supports binding expressions.
     /// </remarks>
     [AutoResolve]
     public string Collection { get; set; }
 
+
+    /// <summary>
+    /// The api key for client authentication, pass null for managed identity.
+    /// </summary>
+    /// <remarks>
+    /// This property supports binding expressions.
+    /// </remarks>
+    [AutoResolve]
+    public string? ApiKey { get; set; }
```

Comment:
Same suggestion about renaming this to `Credentials`. However, in this case, we need to make it clear that this is not the actual credential, but is the *name* of the credential. For that reason, I suggest `CredentialSettingName`.

Reply:
Updated to CredentialSettingName